### PR TITLE
Refactor/cs user

### PR DESCRIPTION
**Pull request by kuenishi**
Moving user-related functions from `riak_cs_utils` to `riak_cs_user`, like `riak_cs_bucket` and `riak_cs_manifest`. Module-per-record code structure will bring more visibility about Riak CS data structure.

This is preliminary refactoring for CS-side caching of user records - while I believe this is useful regardless of usefulness of caching. Discussion and description will follow this pull request. Will be cherry-picked to develop branch.

### CODE REVIEW

File: src/riak_cs_user.erl [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% ---------------------------------------------------------------------
+
+%% @doc riak_cs utility functions
```
Comment (shino) [inline src/riak_cs_user.erl]: This description seems too wide.

File: src/riak_cs_user.erl [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+                    User = update_user_record(hd(Values)),
+                    Buckets = riak_cs_bucket:resolve_buckets(Values, [], KeepDeletedBuckets),
+                    {ok, {User?RCS_USER{buckets=Buckets}, Obj}}
+            end;
+        Error ->
+            Error
+    end.
+
+%% @doc Retrieve a Riak CS user's information based on their
+%% canonical id string.
```
Comment (shino) [inline src/riak_cs_user.erl]: nitpick: this function is used to search user key by email

File: src/riak_cs_user.erl [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+                              [Value, Index, Reason]),
+            Error
+    end.
+
+%% @doc Determine if the specified user account is a system admin.
+-spec is_admin(rcs_user()) -> boolean().
+is_admin(User) ->
+    is_admin(User, riak_cs_config:admin_creds()).
+
+-spec to_3tuple(rcs_user()) -> tuple().
```
Comment (shino) [inline src/riak_cs_user.erl]: (not related this refactoring...)

The name and return type are rather general. `riak_cs.hrl` has type `acl_owner` and `acl_owner3`.

```
%% acl_owner3: {display name, canonical id, key id}
-type acl_owner3() :: {string(), string(), string()}.
-type acl_owner() :: acl_owner2() | acl_owner3().
```

These are not so restrictive but more explanatory (at least for me :) ).

File: src/riak_cs_user.erl [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+
+%% Public API
+-export([
+         create_user/2,
+         create_user/4,
+         display_name/1,
+         is_admin/1,
+         get_user/2,
+         get_user_by_index/3,
+         get_user_index/3,
```
Comment (shino) [inline src/riak_cs_user.erl]: Not used externally?

File: test/riak_cs_utils_test.erl [hunk 5]
```diff
@@ -71,7 +71,7 @@ riak_cs_utils_test_() ->
 name_matches() ->
```
Comment (shino) [inline test/riak_cs_utils_test.erl]: Better to have `riak_cs_user_test.erl` ?

Comment (shino) [inline test/riak_cs_utils_test.erl]: Oops... these functions are NOT unit tests.
Then the question I should ask is: whether should we keep these functions or simply discard them?

Comment (kuenishi) [inline test/riak_cs_utils_test.erl]: Maybe so, but this test is not used. This tests seem old ones, and if we fix this, we'll need Riakc mock behind Riak client and several fixes.

Comment (kuenishi) [inline test/riak_cs_utils_test.erl]: Well, no functions are used and all have replacements in riak_test - I'll remove them.

**Comment by shino** [comment]
Nice refactoring! Good to have the module with specific scope :smile: 

I added some comments but all are ones that exist all along before this PR.
So I can +1 without addressing them by creating a new issue.

**Comment by kuenishi** [comment]
@borshop merge

**Comment by shino** [comment]
Added milestone 1.5.1

**Pull request merged**